src/signing/index.ts

```diff
@@ -370,8 +371,12 @@ export default class SignatureRequestManager {
       // Using BigInt instead solves the Infinity issue, and now allows messages of any length to be
       // signed.
       //
+      let sigToConvert = sigRequest
+      if (!hexHasPrefix(sigToConvert)) {
+        sigToConvert = hexAddPrefix(sigRequest)
+      }
```

Comment:
```suggestion
      const sigToConvert = hexAddPrefix(sigRequest)
```

they wrote it so that you don't need to check, it already does
https://github.com/polkadot-js/common/blob/master/packages/util/src/hex/addPrefix.ts